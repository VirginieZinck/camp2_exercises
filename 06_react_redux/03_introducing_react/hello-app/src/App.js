import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class Hello extends Component {
  render() {
    return (
      <h1>Hello, {this.props.name}</h1>
    )
  }
}

function Hello2(props) {
  return (
    <h1> Hello2, {props.name} </h1>
  );
}

function Hello3(name) {
  return (
    <h1> Hello3, {name}</h1>
  );
}

class App extends Component {
  // constructor(props) {
  //   super(props)
  //   this.state = {counter: 0}
  // }

  constructor(props) {
    super(props)
  }

  render() {
    return (

      <div className="App">
        <header className="App-header">
          <Hello name="Virginie"/>
          <Hello2 name="Virginie"/>
          {Hello3("Virginie")}

        </header>
      </div>
    );
  }



}

export default App;
