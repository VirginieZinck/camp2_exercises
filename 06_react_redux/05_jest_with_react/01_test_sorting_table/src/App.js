import React, { Component } from "react";
import _ from "underscore";

function Row(props) {
  return (
    <tr>
      <td id="td1">{props.decathlon_id}</td>
      <td id="td2">{props.title}</td>
      <td id="td3">{props.price}</td>
    </tr>
  );
}

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      lines: this.props.lines,
      sortedBy: "decathlon_id",
      reverseSort: false
    };
  }

  filter(filterBy) {
    if (this.state.sortedBy === filterBy) {
      this.setState({reverseSort: !this.state.reverseSort});
    } else {
      this.setState({
        sortedBy: filterBy,
        reverseSort: false
      });
    }
  }

  render() {
    let sortedLines = _.sortBy(this.state.lines, this.state.sortedBy);
    if (this.state.reverseSort) {
      sortedLines.reverse();
    }
    return (
      <table>
        <thead>
          <tr>
            <th id="th1" onClick={() => this.filter("decathlon_id")}>ID</th>
            <th id="th2" onClick={() => this.filter("title")}>Title</th>
            <th id="th3" onClick={() => this.filter("price")}>Price</th>
          </tr>
        </thead>
        {sortedLines.map(Row)}
      </table>
    );
  }
}

export default App;
