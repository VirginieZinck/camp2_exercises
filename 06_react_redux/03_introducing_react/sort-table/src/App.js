import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import _ from "underscore";

const products =
[
  { "decathlon_id": 8282689, "title": "Corne chasse 14cm", "price": 9.99 },
  { "decathlon_id": 8354464, "title": "Basic L print Long Gold Fusion", "price": 9.99 },
  { "decathlon_id": 8380024, "title": "RUN ELIOPRIME", "price": 54.99 },
  { "decathlon_id": 8379970, "title": "Pantalon Gym", "price": 12.99 },
  { "decathlon_id": 8247793, "title": "PALMES WADERS", "price": 24.99 },
  { "decathlon_id": 8357549, "title": "MINIMIZER EDEN UNI  NOIR", "price": 19.99 },
  { "decathlon_id": 8326155, "title": "Pantalon Training mesh marine", "price": 44.99 },
  { "decathlon_id": 8329121, "title": "COUTEAU A PALOURDES", "price": 4.99 },
  { "decathlon_id": 8370749, "title": "Doudoune Hike 100 garçon bleu", "price": 9.99 },
  { "decathlon_id": 8298354, "title": "OREILLER CONFORT", "price": 6.99 },
  { "decathlon_id": 8044622, "title": "2 guêtres RIDING noir", "price": 14.99 },
  { "decathlon_id": 8249674, "title": "BOBINE FUN 2 3 4mm X 40 20 12m", "price": 6.99 },
  { "decathlon_id": 8353265, "title": "Justaucorps manche longue Gym.", "price": 34.99 }
]

function Row(props) {
  return (
    <tr>
      <td> {props.product.decathlon_id} </td>
      <td> {props.product.title} </td>
      <td> {props.product.price} </td>
    </tr>
  );
}

class Table extends Component {
  constructor (props) {
    super(props);
    this.state = {products:products};
  }

  sortById () {
    this.setState(
      {products: _.sortBy(this.state.products, 'decathlon_id')}
    )
  }

  sortByTitle () {
    this.setState(
      {products: _.sortBy(this.state.products, 'title')}
    )
  }

  sortByPrice () {
    this.setState(
      {products: _.sortBy(this.state.products, 'price')}
    )
  }

  render() {
    return (
      <table>
        <tr>
          <th>
            Decathlon ID
            <button type="button" onClick={()=>this.sortById()}>Sort!</button>
          </th>
          <th> Title
            <button type="button" onClick={()=>this.sortByTitle()}>Sort!</button>
          </th>
          <th>
            Price
            <button type="button" onClick={()=>this.sortByPrice()}>Sort!</button>
          </th>
        </tr>
        {this.state.products.map( item => {
          return <Row product={item}/>;
        })}
      </table>
    )
  }
}


class App extends Component {
  render() {
    return (
      <Table/>
    );
  }
}

export default App;
