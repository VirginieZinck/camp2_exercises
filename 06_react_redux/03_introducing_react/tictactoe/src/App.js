import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Cell from './Cell';
import tictactoe from './tictactoe';


class App extends Component {
  constructor (props) {
    super(props);
    this.state = {
      grid:tictactoe.emptyGrid,
      nextPlayer:"X"
    };
  }

  hasWinner = () => {
    const isWinningLine = (line) => {
      const pattern =
        line
          .map((coordinate) => this.state.grid[coordinate.letter][coordinate.digit])
          .join("");

      return pattern === "XXX" || pattern === "OOO";
    };

    return tictactoe.WINNING_COORDINATES.some(isWinningLine);
  }

  updateCell = (nextPlayer,row,col) => {
    //update Grid
    let newGrid=this.state.grid;
    newGrid[row][col]=nextPlayer;

    //check if hasWinner
    if (this.hasWinner()) {
      nextPlayer="You Won!!!"
    };

    //update next Player
    if (nextPlayer==="O") {
      nextPlayer="X";
    } else {
      nextPlayer="O";
    }

    this.setState(
      {
        grid:newGrid,
        nextPlayer:nextPlayer
      }
    );
  }

  render() {
    return (
      <div className="App">
          <h1 className="App-title pb-3">Tic-Tac-Toe</h1>
          <h5 className="pb-3">Next player : {this.state.nextPlayer}</h5>
          <div className="container">
            <div className="row">
                <Cell row="a" col="0" nextPlayer={this.state.nextPlayer} value={this.state.grid.a[0]} function={this.updateCell}/>
                <Cell row="a" col="1" nextPlayer={this.state.nextPlayer} value={this.state.grid.a[1]} function={this.updateCell}/>
                <Cell row="a" col="2" nextPlayer={this.state.nextPlayer} value={this.state.grid.a[2]} function={this.updateCell}/>
            </div>
            <div className="row">
                <Cell row="b" col="0" nextPlayer={this.state.nextPlayer} value={this.state.grid.b[0]} function={this.updateCell}/>
                <Cell row="b" col="1" nextPlayer={this.state.nextPlayer} value={this.state.grid.b[1]} function={this.updateCell}/>
                <Cell row="b" col="2" nextPlayer={this.state.nextPlayer} value={this.state.grid.b[2]} function={this.updateCell}/>
            </div>
            <div className="row">
                <Cell row="c" col="0" nextPlayer={this.state.nextPlayer} value={this.state.grid.c[0]} function={this.updateCell}/>
                <Cell row="c" col="1" nextPlayer={this.state.nextPlayer} value={this.state.grid.c[1]} function={this.updateCell}/>
                <Cell row="c" col="2" nextPlayer={this.state.nextPlayer} value={this.state.grid.c[2]} function={this.updateCell}/>
            </div>
          </div>
      </div>
    );
  }
}

export default App;
