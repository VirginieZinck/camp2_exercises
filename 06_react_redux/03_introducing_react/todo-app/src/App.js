import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Row from './Row';
import TaskForm from './TaskForm';


class ToDoList extends Component {
  constructor (props) {
    super(props);
    this.state = {list:[
      {status:"Done",label:"Faire mon EAD"},
      {status:"ToDo",label:"Dentiste"}
    ]};
  }

  checked = (item) => {
    return item.status==="Done";
  }

  checkTask = (item,index) => {
    let newList = this.state.list;

    if (newList[index].status === "Done") {
      newList[index].status = "Todo";
    } else {
    newList[index].status = "Done";
    }

    this.setState({list: newList}, () => console.log(this.state))
  }

  createTask = (label) => {
    let newList = this.state.list;
    newList.push({status:"Todo",label:label});

    this.setState({list: newList}, () => console.log(this.state))
  }

  deleteTask = (index) => {
    let newList = this.state.list;
    newList.splice(index,1);

    this.setState({list: newList}, () => console.log(this.state))

  }

  render() {
    return (
      <div className="container">

        <TaskForm function={this.createTask}/>

        <div className="row">
            <th> Tasks </th>
        </div>

        {this.state.list.map( (item,index)=>{
          return (
            <Row
              item={item}
              index={index}
              checked={this.checked}
              onCheck={this.checkTask}
              onDelete={this.deleteTask}
            />
          );
        })}
      </div>
    )
  }
}


class App extends Component {
  render() {
    return (
      <div className="App">
        <h1 className="App-title">ToDo List</h1>
        <ToDoList/>
      </div>
    );
  }
}

export default App;
