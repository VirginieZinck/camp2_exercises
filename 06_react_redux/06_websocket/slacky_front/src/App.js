import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import ConnectionForm from './ConnectionForm';
import MsgTable from './MsgTable';
import MsgForm from './MsgForm';

class App extends Component {

  constructor(props) {
    super(props);
    this.ws = new WebSocket("ws://localhost:3001");
    this.state = {
      connected: 0,
      userName:"",
      msgArray:[],
      userArray:[]
    };
  }

  componentDidMount() {

    // What to do when we receive a message from server?
    this.ws.onmessage = (event) => {

      const msgReceived=JSON.parse(event.data);
      console.log("Message: " + msgReceived);

      if (msgReceived.type==="MSG") {
        //we received a new chat : we add it to msgArray for display
        const newMsgArray = this.state.msgArray;
        newMsgArray.push({
          userName:msgReceived.data,
          msg:msgReceived.msg
        });

        this.setState({
          msgArray:newMsgArray
        },() => {
          console.log(`New message added ${msgReceived.msg} from ${msgReceived.userName}.`);
        });

      } else if (msgReceived.type==="USRLIST"){
        //we received an updated user list
        const newUserArray = msgReceived.UserList;

        this.setState({
          userArray:newUserArray
        },() => {
          console.log(`User list updated ${msgReceived.UserList}.`);
        });

      } else {
        console.warn("invalid message type", msgReceived.type);
      }
    };

    // Alert the server that the client is gone
    window.addEventListener("beforeunload", () => {

      const msgSent={
        type:"CLOSEUSR",
        userName:this.state.userName
      };
      this.ws.send(JSON.stringify(msgSent));
    });
  }

  connect = (userName) => {
    const newUserName = userName;
    console.log("newuserName",newUserName);

    const msgSent={
      type:"NEWUSR",
      userName:newUserName
    };

    this.setState({
      connected: 1,
      userName:newUserName
    },() => {
      // We send a message to server when we connect
      console.log(`User ${this.state.userName} is connected.`);
      this.ws.send(JSON.stringify(msgSent));
    });
  }

  createMsg = (msg) => {
    console.log("newMsg",msg);

    const newMsgArray = this.state.msgArray;
    newMsgArray.push({
      userName:this.state.userName,
      msg:msg
    });


    const msgSent={
      type:"MSG",
      msg:msg,
      userName:this.state.userName
    };

    this.setState({
      msgArray:newMsgArray
    },() => {
      // We send a message to server when we connect
      console.log(`New message added ${msg}.`);
      this.ws.send(JSON.stringify(msgSent));
    });
  }

  render() {
    return (
      <div className="App">
        <h1 className="App-title pb-5">Welcome to Slacky</h1>
        <ConnectionForm
          onSubmit={this.connect}
          connected={this.state.connected}
          userName={this.state.userName}
        />
        <div className="container ">
          <div id="msg" className="row justify-content-center">
            <h4>Messages</h4>
          </div>
          <MsgTable msgArray={this.state.msgArray}/>
        </div>
        <MsgForm connected={this.state.connected} onSubmit={this.createMsg}/>
      </div>
    );
  }
}

export default App;
