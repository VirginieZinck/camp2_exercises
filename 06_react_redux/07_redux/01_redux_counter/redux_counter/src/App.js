import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { connect } from "react-redux";


class App extends Component {
  render() {
    console.log("App props",this.props);
    return (
      <div className="App">
        <img src={logo} className="App-logo" alt="logo" />
        <h1 className="App-title">Counter in React Redux</h1>
        <button onClick={this.props.decrement}> - </button>
        <h4> {this.props.counter} </h4>
        <button onClick={this.props.increment}> + </button>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    counter: state.counter
  }
}

function mapDispatchToProps(dispatch) {
  return {
    increment: () => dispatch({type: "INCREMENT"}),
    decrement: () => dispatch({type: "DECREMENT"})
  }
}

const ConnectedApp = connect(mapStateToProps, mapDispatchToProps)(App);

export default ConnectedApp;
