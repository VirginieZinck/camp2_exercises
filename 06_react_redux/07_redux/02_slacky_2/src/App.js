import React, { Component } from "react";
import "./App.css";
import ConnectedLogin from "./Login";
import ConnectedChat from "./Chat";
import { connect } from 'react-redux';


class App extends Component {

  componentDidMount() {
    // Listen for messages
    this.props.websocket.addEventListener("message", event => {
      const message = JSON.parse(event.data);
      console.log("Message from server ", message);
      switch (message.type) {
        case "CONNECTION_START":
        default:
          return;
        case "MESSAGES":
          // this.setState({ messages: message.data });
          this.props.updateMessages(message.data);
          return;
      }
    });
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Slacky</h1>
        </header>

        {this.props.userName ? (
          <ConnectedChat />
        ) : (
          <ConnectedLogin />
        )}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    userName: state.userName,
    websocket: state.websocket

  }
}

function mapDispatchToProps(dispatch) {
  return {
    updateMessages: (messages) => dispatch({type: "UPDATEMESSAGES",messages:messages})
  }
}

const ConnectedApp = connect(mapStateToProps, mapDispatchToProps)(App);

export default ConnectedApp;
