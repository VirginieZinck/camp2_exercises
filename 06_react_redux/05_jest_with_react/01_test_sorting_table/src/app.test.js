import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import { configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Wapiti from 'wapiti';

const fakeProducts =
[
  { "decathlon_id": 8282689, "title": "Corne chasse 14cm", "price": 9.99 },
  { "decathlon_id": 8354464, "title": "Basic L print Long Gold Fusion", "price": 9.99 },
  { "decathlon_id": 8380024, "title": "RUN ELIOPRIME", "price": 54.99 }
]


beforeAll(() => {
  configure({ adapter: new Adapter() });
})

//initial react test
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

//snapshop test on initial render
test('Snapshot test on initial render', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

//snapshop test on initial render
test('Snapshot test on initial render with props', () => {
  const component = renderer.create(<App lines={fakeProducts} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Get first ID at first render (default sort by ID)', () => {
  // Render the page by default sort
  const app = shallow(<App lines={fakeProducts}/>);
  expect(app.find('td').at(0).text()).toEqual('8282689');
});

test('Get first ID after one sort by Id (sort desc)', () => {
  // Render the page by default sort
  const app = shallow(<App lines={fakeProducts}/>);

  expect(app.find('td').at(0).text()).toEqual('8282689');
  app.find('th').at(0).simulate('click');
  expect(app.find('td').at(0).text()).toEqual('8380024');
});

test('Get first ID after 2nd sort by Id (sort asc)', () => {
  // Render the page by default sort
  const app = shallow(<App lines={fakeProducts}/>);
  expect(app.find('td').at(0).text()).toEqual('8282689');
  app.find('th').at(0).simulate('click');
  expect(app.find('td').at(0).text()).toEqual('8380024');
  app.find('th').at(0).simulate('click');
  expect(app.find('td').at(0).text()).toEqual('8282689');
});

test('Get first title after one sort by Title (sort asc)', () => {
  // Render the page by default sort
  const app = shallow(<App lines={fakeProducts}/>);
  app.find('th').at(1).simulate('click');
  expect(app.find('td').at(1).text()).toEqual('Basic L print Long Gold Fusion');
});

test('Get first title after 2nd sort by Title (sort desc)', () => {
  // Render the page by default sort
  const app = shallow(<App lines={fakeProducts}/>);
  app.find('th').at(1).simulate('click');
  expect(app.find('td').at(1).text()).toEqual('Basic L print Long Gold Fusion');
  app.find('th').at(1).simulate('click');
  expect(app.find('td').at(1).text()).toEqual('RUN ELIOPRIME');
});

test('Get first price after one sort by price (sort asc)', () => {
  // Render the page by default sort
  const app = shallow(<App lines={fakeProducts}/>);
  app.find('th').at(2).simulate('click');
  expect(app.find('td').at(2).text()).toEqual("9.99");
});

test('Get first title after 2nd sort by Title (sort desc)', () => {
  // Render the page by default sort
  const app = shallow(<App lines={fakeProducts}/>);
  app.find('th').at(2).simulate('click');
  expect(app.find('td').at(2).text()).toEqual("9.99");
  app.find('th').at(2).simulate('click');
  expect(app.find('td').at(2).text()).toEqual("54.99");
});


//Wapiti tests for integration tests
test("Test first render & click on sort by ID once", () => {
  expect.assertions(1);

  return Wapiti.goto("http://localhost:3000")
    .capture(() => document.querySelector("#td1").textContent)
    .click("#th1")
    .capture(() => document.querySelector("#td1").textContent)
    .run()
    .then(result => {
      expect(result).toEqual(["8044622", "8380024"]);
    });
});

//Wapiti tests for integration tests
test("Test full scenario with sorts on all columns", () => {
  expect.assertions(1);

  return Wapiti.goto("http://localhost:3000")
    .capture(() => document.querySelector("#td1").textContent)
    .click("#th1")
    .capture(() => document.querySelector("#td1").textContent)
    .click("#th1")
    .capture(() => document.querySelector("#td1").textContent)
    .click("#th2")
    .capture(() => document.querySelector("#td2").textContent)
    .click("#th2")
    .capture(() => document.querySelector("#td2").textContent)
    .click("#th3")
    .capture(() => document.querySelector("#td3").textContent)
    .click("#th3")
    .capture(() => document.querySelector("#td3").textContent)
    .run()
    .then(result => {
      expect(result).toEqual([
        "8044622",
        "8380024",
        "8044622",
        "2 guÃªtres RIDING noir",
        "RUN ELIOPRIME",
        "4.99",
        "54.99"
      ]);
    });
});
