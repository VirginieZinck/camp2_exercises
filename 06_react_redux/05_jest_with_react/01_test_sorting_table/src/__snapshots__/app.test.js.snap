// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test on initial render 1`] = `
<table>
  <thead>
    <tr>
      <th
        id="th1"
        onClick={[Function]}
      >
        ID
      </th>
      <th
        id="th2"
        onClick={[Function]}
      >
        Title
      </th>
      <th
        id="th3"
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
</table>
`;

exports[`Snapshot test on initial render with props 1`] = `
<table>
  <thead>
    <tr>
      <th
        id="th1"
        onClick={[Function]}
      >
        ID
      </th>
      <th
        id="th2"
        onClick={[Function]}
      >
        Title
      </th>
      <th
        id="th3"
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
  <tr>
    <td
      id="td1"
    >
      8282689
    </td>
    <td
      id="td2"
    >
      Corne chasse 14cm
    </td>
    <td
      id="td3"
    >
      9.99
    </td>
  </tr>
  <tr>
    <td
      id="td1"
    >
      8354464
    </td>
    <td
      id="td2"
    >
      Basic L print Long Gold Fusion
    </td>
    <td
      id="td3"
    >
      9.99
    </td>
  </tr>
  <tr>
    <td
      id="td1"
    >
      8380024
    </td>
    <td
      id="td2"
    >
      RUN ELIOPRIME
    </td>
    <td
      id="td3"
    >
      54.99
    </td>
  </tr>
</table>
`;
